python/eos/figure/plot.py

```diff
@@ -108,9 +109,15 @@ def draw(self, ax):
             ax.xaxis.set_minor_formatter(plt.NullFormatter())
             ax.xaxis.set_tick_params(bottom=False, top=False)
         else:
-            ax.xaxis.set_major_locator(matplotlib.ticker.AutoLocator())
+            if not self.log_scale:
```

Comment:
Can we not determine this condition from the ``ax.xaxis`` object and dispense with adding a new field that contains information recorded elsewhere?

The matplotlib documentation suggests that ``ax.xaxis.get_scale()`` returns ``linear`` and ``log`` for the scales that we commonly use.